spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/DeleteOrphanFilesProcedure.java

```diff
@@ -37,11 +37,11 @@
 import org.apache.spark.unsafe.types.UTF8String;
 
 /**
- * A procedure that removes orphan files in a table.
+ * A procedure that deletes orphan files in a table.
  *
  * @see SparkActions#deleteOrphanFiles(Table)
  */
-public class RemoveOrphanFilesProcedure extends BaseProcedure {
+public class DeleteOrphanFilesProcedure extends BaseProcedure {
```

Comment:
It seems like this is a needless breaking API change. Same with the procedure rename. Can we avoid breaking APIs and just rename what we can safely?

Reply:
Agree. 

But I thought it is better to make Action name and procedure name same, so that user will not get confused. 
Even though it is breaking change we can add it in the release notes and notify users ? 

@rdblue , @RussellSpitzer : If you still think it is ok to have different names for procedure and Action, I can revert this. WDYT ?

Reply:
Can we have multiple names for a procedure? Now that I'm looking at this, I don't think we should consider the procedure's class name something that we can't change, just the name that we use to call it in SQL.

Reply:
yeah, I can point both `remove_orphan_files` and `delete_orphan_files` to use `DeleteOrphanFilesProcedure`
to avoid breaking change. Documentation also I can add a note for the same.

Reply:
I actually thought `remove` is a better verb than `delete`  for user-facing procedures because `delete` is used in many other contexts of delete files and merge-on-read. Is it necessary to match the java class and procedure name?

Reply:
I thought, if the procedure name and Action name is same, user don't have to have hard time remembering the names. 
Thats the only reason why I changed.